### ISSUE TITLE
[a11y]: ARIA Specification violations for SideNavMenu and SideNavMenuItem of UI Shell

### ISSUE THREAD

**Issue by vgowrabathuni** [carbon-design-system/carbon-labs#1210]
### Browser

Chrome

### Operating System

MacOS

### Package version

v0.87.0

### React version

React v18.3.1

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker - Latest Deployment

### Description

Currently [SideNavMenu](https://github.com/carbon-design-system/carbon-labs/blob/5ba286737369a35f3391af93d5dc115eb69783da/packages/react/src/components/UIShell/components/SideNavMenu.tsx#L486) and [SideNavMenuItem](https://github.com/carbon-design-system/carbon-labs/blob/5ba286737369a35f3391af93d5dc115eb69783da/packages/react/src/components/UIShell/components/SideNavMenuItem.tsx#L94) of UI Shell are generating below violations

ARIA Specification
- The ARIA attributes "aria-expanded" are not valid for the element `<li>` with implicit ARIA role "generic"

```
<li
    aria-describedby="tooltip-:ra9:"
    id="side-nav-menu-:ra8:"
    class="cds--side-nav__item cds--side-nav__item--icon"
    aria-expanded="false">
```
- The ARIA attributes "aria-selected" are not valid for the element <a> with implicit ARIA role "link"

```
<a
    tabindex="0"
    class="cds--side-nav__link"
    aria-selected="false"
    depth="1"
    href="#">
```

### WCAG 2.1 Violation

ARIA Specification

### Reproduction/example

https://labs.carbondesignsystem.com/react/iframe.html?globals=&args=&id=components-uishell--demo&viewMode=story

### Steps to reproduce

- The ARIA attributes "aria-expanded" are not valid for the element <li> with implicit ARIA role "generic"

    Scan the page with IBM Equal accessibility checker

<img width="1719" height="977" alt="Image" src="https://github.com/user-attachments/assets/91cf9494-dfaa-4053-adf6-ace2b6a3da79" />

- The ARIA attributes "aria-selected" are not valid for the element with implicit ARIA role "link"
  
   Expand the Home menu and then scan with the IBM Equal accessibility checker

<img width="1719" height="977" alt="Image" src="https://github.com/user-attachments/assets/39724f40-8669-460b-b920-99f00e0139f3" />

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-labs/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon-labs/issues) for duplicate problems
